File: golang/libs/events_system/registered_listener.go
package events_system

import "reflect"

type RegisteredListener struct {
	listener      Listener
	eventExecutor EventExecutor
}

func NewRegisteredListener(listener Listener, eventExecutor EventExecutor) RegisteredListener {
	return RegisteredListener{
		listener:      listener,
		eventExecutor: eventExecutor,
	}
}

func (registeredListener *RegisteredListener) CallEvent(event *Event) {
	if f := reflect.ValueOf(event).FieldByName("Cancel"); f == reflect.ValueOf(nil) || !f.Bool() {
		registeredListener.eventExecutor(registeredListener.listener, event)
	}
}

File: java/libs/[ITS] EventSystem/src/net/kio/its/event/RegisteredListener.java
package net.kio.its.event;

public class RegisteredListener {

    private final Listener listener;
    private final EventPriority eventPriority;
    private final EventExecutor executor;

    public RegisteredListener(Listener listener, EventPriority eventPriority, EventExecutor eventExecutor) {
        this.listener = listener;
        this.eventPriority = eventPriority;
        this.executor = eventExecutor;
    }

    public EventPriority getEventPriority() {
        return eventPriority;
    }

    public void callEvent(Event event) {
        if (!(event instanceof Cancellable) || !(((Cancellable) event).isCancelled())) {
            executor.execute(listener, event);
        }
    }
}
